catroid/src/main/java/org/catrobat/catroid/ui/dialogs/LegoSensorPortConfigDialog.java

```diff
@@ -28,160 +28,112 @@
-					nxtSensor = NXTSensor.Sensor.NO_SENSOR;
-			}
-		} else {
-			switch (clickedItem) {
-				case R.string.formula_editor_sensor_lego_ev3_sensor_touch:
-					title = getString(R.string.ev3_sensor_touch);
-					ev3Sensor = EV3Sensor.Sensor.TOUCH;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_infrared:
-					title = getString(R.string.ev3_sensor_infrared);
-					ev3Sensor = EV3Sensor.Sensor.INFRARED;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color:
-					title = getString(R.string.ev3_sensor_color);
-					ev3Sensor = EV3Sensor.Sensor.COLOR;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color_ambient:
-					title = getString(R.string.ev3_sensor_color_ambient);
-					ev3Sensor = EV3Sensor.Sensor.COLOR_AMBIENT;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color_reflected:
-					title = getString(R.string.ev3_sensor_color_reflected);
-					ev3Sensor = EV3Sensor.Sensor.COLOR_REFLECT;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_hitechnic_color:
-					title = getString(R.string.ev3_sensor_hitechnic_color);
-					ev3Sensor = EV3Sensor.Sensor.HT_NXT_COLOR;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_nxt_temperature_c:
-					title = getString(R.string.ev3_sensor_nxt_temperature_c);
-					ev3Sensor = EV3Sensor.Sensor.NXT_TEMPERATURE_C;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_nxt_temperature_f:
-					title = getString(R.string.ev3_sensor_nxt_temperature_f);
-					ev3Sensor = EV3Sensor.Sensor.NXT_TEMPERATURE_F;
-					break;
-				default:
-					title = getString(R.string.nxt_sensor_not_found);
-					ev3Sensor = EV3Sensor.Sensor.NO_SENSOR;
-			}
-		}
+	private SensorInfo getSensorInfo(int clickedItem, Lego type) {
+		SensorInfo info = sensorInfoMap.get(clickedItem);
+		Enum sensor = type == Lego.NXT ? NXTSensor.Sensor.NO_SENSOR : EV3Sensor.Sensor.NO_SENSOR;
+		return info != null ? info : new SensorInfo(getString(R.string.nxt_sensor_not_found), sensor);
 	}
 
 	private void checkIfOverwrite(final int selectedPort, String selected, final Lego legoType) {
-		final Context dialogueContext = getActivity();
+		final Context dialogContext = getActivity();
 		if (selected.equals(getString(R.string.nxt_no_sensor))) { // nxt_no_sensor equals ev3_no_sensor
-			overwrite(dialogueContext, selectedPort, legoType);
-		} else if (!selected.equals(title)) {
-			AlertDialog.Builder builder = new AlertDialog.Builder(dialogueContext);
+			overwriteSensorPortConfig(dialogContext, selectedPort, legoType);
+		} else if (!selected.equals(sensorInfo.title)) {
+			AlertDialog.Builder builder = new AlertDialog.Builder(dialogContext);
 			builder.setTitle(R.string.lego_nxt_overwrite_sensor_port_dialog_title)
 					.setMessage(R.string.lego_nxt_overwrite_sensor_port_dialog_message)
 					.setNegativeButton(R.string.no, new DialogInterface.OnClickListener() {
```

Comment:
If you don't want to do anything special in onDismiss i.e. if you rely on onDismiss being called when the dialog is closed via this button just replace the DialogInterface.OnClickListener() with null.